File: pepper-farm/backend/models/farm_zone.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Numeric, ForeignKey
from sqlalchemy.sql import func
from database import Base

class FarmZone(Base):
    __tablename__ = "FarmZones"

    ZoneId           = Column(Integer, primary_key=True, autoincrement=True)
    ZoneName         = Column(String(100), nullable=False, unique=True)
    ZoneCode         = Column(String(50), nullable=True, unique=True)
    PepperId         = Column(Integer, ForeignKey("PepperVarieties.PepperId"), nullable=True)
    AreaSquareMeters = Column(Numeric(10, 2), nullable=True)
    Latitude         = Column(Numeric(9, 6), nullable=True)
    Longitude        = Column(Numeric(9, 6), nullable=True)
    SoilType         = Column(String(100), nullable=True)
    IrrigationMethod = Column(String(100), nullable=True)
    Notes            = Column(String(500), nullable=True)
    IsActive         = Column(Boolean, nullable=False, default=True)
    CreatedAt        = Column(DateTime, nullable=False, server_default=func.sysutcdatetime())

File: pepper-farm/backend/Models/FarmZone.cs
namespace PepperFarm.Api.Models;

public class FarmZone
{
    public int ZoneId { get; set; }
    public string ZoneName { get; set; } = string.Empty;
    public string? ZoneCode { get; set; }
    public int? PepperId { get; set; }
    public decimal? AreaSquareMeters { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    public string? SoilType { get; set; }
    public string? IrrigationMethod { get; set; }
    public string? Notes { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public PepperVariety? Pepper { get; set; }
}
